source/GameInterface/Services/MobileParties/Patches/MobilePartyFieldPatches.cs

```diff
@@ -0,0 +1,892 @@
+    }
+
+    [HarmonyTranspiler]
+    private static IEnumerable<CodeInstruction> PartyTradeGoldTranspiler(IEnumerable<CodeInstruction> instructions)
+    {
+        var field = AccessTools.Field(typeof(MobileParty), nameof(MobileParty._partyTradeGold));
+        var fieldIntercept = AccessTools.Method(typeof(MobilePartyFieldPatches), nameof(PartyTradeGoldIntercept));
+
+        foreach (var instruction in instructions)
+        {
+            if (instruction.StoresField(field))
+            {
+                yield return new CodeInstruction(OpCodes.Call, fieldIntercept);
+            }
+            else
+            {
+                yield return instruction;
+            }
+        }
+    }
+
+    public static void PartyTradeGoldIntercept(MobileParty instance, int newPartyTradeGold)
+    {
+        if (CallOriginalPolicy.IsOriginalAllowed())
+        {
+            instance._partyTradeGold = newPartyTradeGold;
+            return;
+        }
+        if (ModInformation.IsClient)
+        {
+            Logger.Error("Client added unmanaged item: {callstack}", Environment.StackTrace);
+            instance._partyTradeGold = newPartyTradeGold;
+            return;
+        }
+
+        MessageBroker.Instance.Publish(instance, new PartyTradeGoldChanged(newPartyTradeGold, instance.StringId));
+
+        instance._partyTradeGold = newPartyTradeGold;
+    }
+
+    [HarmonyTranspiler]
+    private static IEnumerable<CodeInstruction> IgnoredUntilTimeTranspiler(IEnumerable<CodeInstruction> instructions)
+    {
+        var field = AccessTools.Field(typeof(MobileParty), nameof(MobileParty._ignoredUntilTime));
+        var fieldIntercept = AccessTools.Method(typeof(MobilePartyFieldPatches), nameof(IgnoredUntilTimeIntercept));
+
+        foreach (var instruction in instructions)
+        {
+            if (instruction.StoresField(field))
+            {
+                yield return new CodeInstruction(OpCodes.Call, fieldIntercept);
+            }
+            else
+            {
+                yield return instruction;
+            }
+        }
+    }
+
+    public static void IgnoredUntilTimeIntercept(MobileParty instance, CampaignTime newIgnoredUntilTime)
+    {
+        if (CallOriginalPolicy.IsOriginalAllowed())
+        {
+            instance._ignoredUntilTime = newIgnoredUntilTime;
+            return;
+        }
+        if (ModInformation.IsClient)
+        {
+            Logger.Error("Client added unmanaged item: {callstack}", Environment.StackTrace);
+            instance._ignoredUntilTime = newIgnoredUntilTime;
+            return;
+        }
+
+        MessageBroker.Instance.Publish(instance, new IgnoredUntilTimeChanged(newIgnoredUntilTime.NumTicks, instance.StringId));
+
+        instance._ignoredUntilTime = newIgnoredUntilTime;
+    }
+
+    [HarmonyTranspiler]
+    private static IEnumerable<CodeInstruction> AverageFleeTargetDirectionTranspiler(IEnumerable<CodeInstruction> instructions)
```

Comment:
This is assigned in the MobilePartyAi.GetFleeBehavior which is not a part of the target methods